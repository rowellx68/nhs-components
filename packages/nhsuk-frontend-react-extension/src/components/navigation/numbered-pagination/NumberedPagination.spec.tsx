import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, { Default } from './NumberedPagination.stories';

it('should render the NumberedPagination component with default props', () => {
  const NumberedPagination = composeStory(Default, meta);
  const { container } = render(<NumberedPagination />);

  expect(container).toMatchSnapshot();
  expect(container.querySelector('nav')).toBeInTheDocument();
});
