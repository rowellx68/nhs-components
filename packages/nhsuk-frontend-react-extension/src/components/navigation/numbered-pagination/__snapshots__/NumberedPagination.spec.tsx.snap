// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the NumberedPagination component with default props 1`] = `
<div>
  <nav
    aria-label="results"
    class="nhsuk-numbered-pagination"
    role="navigation"
  >
    <div
      class="nhsuk-numbered-pagination__prev"
    >
      <a
        class="nhsuk-link nhsuk-numbered-pagination__link"
      >
        <svg
          aria-hidden="true"
          class="nhsuk-numbered-pagination__icon nhsuk-numbered-pagination__icon--prev"
          focusable="false"
          height="13"
          viewBox="0 0 15 13"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z"
          />
        </svg>
        <span
          class="nhsuk-numbered-pagination__link-title"
        >
          Previous
        </span>
      </a>
    </div>
    <ul
      class="nhsuk-numbered-pagination__list"
    >
      <li
        class="nhsuk-numbered-pagination__item nhsuk-numbered-pagination__item--current"
      >
        <a
          aria-current="page"
          aria-label="Page 1"
          class="nhsuk-link nhsuk-numbered-pagination__link"
        >
          1
        </a>
      </li>
      <li
        class="nhsuk-numbered-pagination__item"
      >
        <a
          aria-label="Page 2"
          class="nhsuk-link nhsuk-numbered-pagination__link"
        >
          2
        </a>
      </li>
      <li
        class="nhsuk-numbered-pagination__item nhsuk-numbered-pagination__item--ellipses"
      >
        â‹¯
      </li>
      <li
        class="nhsuk-numbered-pagination__item"
      >
        <a
          aria-label="Page 4"
          class="nhsuk-link nhsuk-numbered-pagination__link"
        >
          4
        </a>
      </li>
      <li
        class="nhsuk-numbered-pagination__item"
      >
        <a
          aria-label="Page 5"
          class="nhsuk-link nhsuk-numbered-pagination__link"
        >
          5
        </a>
      </li>
    </ul>
    <div
      class="nhsuk-numbered-pagination__next"
    >
      <a
        class="nhsuk-link nhsuk-numbered-pagination__link"
      >
        <span
          class="nhsuk-numbered-pagination__link-title"
        >
          Next
        </span>
        <svg
          aria-hidden="true"
          class="nhsuk-numbered-pagination__icon nhsuk-numbered-pagination__icon--next"
          focusable="false"
          height="13"
          viewBox="0 0 15 13"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m8.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z"
          />
        </svg>
      </a>
    </div>
  </nav>
</div>
`;
